packages/visualizations/src/components/Map/utils.js

```diff
@@ -1,19 +1,40 @@
 import chroma from 'chroma-js';
 import geoViewport from '@mapbox/geo-viewport';
 
-export const colorShapes = (geoJson, values, colorScale) => {
+export const colorShapes = (geoJson, values, colorsScale) => {
     // Key in the values is "x"
     // Key in the shapes is "key"
     // We add a color property in the JSON
+    // FIXME Do we always want to take y or accept other keys
     const rawValues = values.map((v) => v.y);
     const min = Math.min(...rawValues);
     const max = Math.max(...rawValues);
-
-    // For now the colorscale is a single color, and we build a scale from it. TBD
-    const colorMin = chroma(colorScale).darken(4).hex();
-    const colorMax = chroma(colorScale).brighten(4).hex();
-
-    const scale = chroma.scale([colorMin, colorMax]).domain([min, max]);
+    let colorMin;
+    let colorMax;
+    let scale;
+
+    if (colorsScale?.type === 'palette') {
+        const tresholdArray = [];
+        colorsScale.colors.forEach((color, i) => {
+            if (i === 0) {
+                tresholdArray.push(min);
+                tresholdArray.push(min + (max - min) / colorsScale.colors.length);
+            } else if (i === colorsScale.colors.length - 1) {
+                tresholdArray.push(max);
+            } else {
+                tresholdArray.push(min + ((max - min) / colorsScale.colors.length) * (i + 1));
+            }
+        });
+        scale = chroma.scale(colorsScale.colors).classes(tresholdArray);
+    } else if (colorsScale?.type === 'gradient') {
+        colorMin = chroma(colorsScale.colors.start).hex();
+        colorMax = chroma(colorsScale.colors.end).hex();
+        scale = chroma.scale([colorMin, colorMax]).domain([min, max]);
+    } else {
+        // Default is basic grey color
+        const uniqueColor = chroma('#CBD2DB').hex();
+        scale = chroma.scale([uniqueColor, uniqueColor]).domain([min, max]);
```

Comment:
I'm a bit confused because I would have expected this code to just return the same color for min and max 😅 

I think it may be easier to read (and possibly type later) if we just used a default value higher in the code (e.g. in MapRender, or in Choropleth if we want different default for different viz types later), like "if colorsScale is not defined, then we use a gradient between light grey and dark grey".